### PR TITLE
[CROSSDATA-440] JAR files should be added to the Spark driver classpath

### PR DESCRIPTION
**Pull request by jjlopezm**

### CODE REVIEW

File: server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala [hunk 1]
```diff
@@ -135,6 +144,26 @@ class ServerActor(cluster: Cluster, xdContext: XDContext, config: ServerActorCon
       st.jobsById.get(JobId(requester, queryId)).get ! CancelJob
   }
 
+  private def addToClasspath(file : File) : Unit = {
+    if (file.exists) {
+      val method : Method = classOf[URLClassLoader].getDeclaredMethod("addURL", classOf[URL])
+
+      method.setAccessible(true)
```
Comment (miguel0afd) [inline server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala]: 😱  but necessaryy

File: testsIT/src/test/scala/com/stratio/crossdata/driver/DriverIT.scala [hunk 2]
```diff
@@ -91,41 +91,41 @@ class DriverIT extends EndToEndTest {
...
-//
-//  it should "be able to execute ADD JAR Command of any HDFS file" in {
-//
-//    val driver = Driver.getOrCreate()
-//    val result = driver.addJar(s"hdfs://repo/file.jar").waitForResult()
-//    driver.stop()
-//
-//    result.hasError should equal (false)
-//  }
+  //  // TODO move to examples??
```
Comment (miguel0afd) [inline testsIT/src/test/scala/com/stratio/crossdata/driver/DriverIT.scala]: Looks like a good moment to move it to examples...

**Comment by miguel0afd** [comment]
Well done!

File: server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala (hunk 1 above)
Comment (pfcoperez) [inline server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala]: Indeed :disappointed:

File: server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala [hunk 3]
```diff
@@ -135,6 +144,26 @@ class ServerActor(cluster: Cluster, xdContext: XDContext, config: ServerActorCon
...
+      method.invoke(ClassLoader.getSystemClassLoader, file.toURI.toURL)
+    } else {
+      logger.warn(s"The file ${file.getName} not exists.")
```
Comment (pfcoperez) [inline server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala]: Shouldn't this raise a catastrophic error?

**Comment by miguel0afd** [comment]
By the way, is the doc up-to-date with this new feature?

File: server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala [hunk 4]
```diff
@@ -135,6 +144,26 @@ class ServerActor(cluster: Cluster, xdContext: XDContext, config: ServerActorCon
       st.jobsById.get(JobId(requester, queryId)).get ! CancelJob
   }
 
+  private def addToClasspath(file : File) : Unit = {
+    if (file.exists) {
+      val method : Method = classOf[URLClassLoader].getDeclaredMethod("addURL", classOf[URL])
+
+      method.setAccessible(true)
+      method.invoke(ClassLoader.getSystemClassLoader, file.toURI.toURL)
```
Comment (pfcoperez) [inline server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala]: You might like to restore reflection accessibility flag after the "dirty work" has been finished `method.setAccessible(false)`

**Comment by pfcoperez** [comment]
+1

File: testsIT/src/test/scala/com/stratio/crossdata/driver/DriverIT.scala [hunk 5]
```diff
@@ -75,7 +74,8 @@ class DriverIT extends EndToEndTest {
 
   "Crossdata Driver" should "be able to close the connection and start it again" in {
     assumeCrossdataUpAndRunning
-    var driver = Driver.getOrCreate(); Driver.getOrCreate()
+    var driver = Driver.getOrCreate();
```
Comment (darroyocazorla) [inline testsIT/src/test/scala/com/stratio/crossdata/driver/DriverIT.scala]: val driver = Driver.getOrCreate() should be enough

Comment (jjlopezm) [inline testsIT/src/test/scala/com/stratio/crossdata/driver/DriverIT.scala]: The test  check  that the driver can be reused, closing and opening again, so I think that it is necessary to be a var. But that problem it's not related with this PR

File: server/src/main/scala/com/stratio/crossdata/server/CrossdataServer.scala [hunk 6]
```diff
@@ -82,14 +82,17 @@ class CrossdataServer extends Daemon with ServerConfig {
             xdContext.getOrElse(throw new RuntimeException("Crossdata context cannot be started")),
             serverActorConfig)),
         actorName)
+      val broadcastActor=actorSystem.actorOf(ResourceManagerActor.props(Cluster(actorSystem),xdContext.get))
```
Comment (pfcoperez) [inline server/src/main/scala/com/stratio/crossdata/server/CrossdataServer.scala]: This is silly but, wouldn't `broadcastActor` be better identified as `resourceMgrActor` or something of the kind?

**Comment by pfcoperez** [comment]
+1

**Pull request merged**